Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/TextBoxLikePropertyCtrl.hxx

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <AzCore/std/limits.h>
+#include <AzCore/std/optional.h>
+
+namespace AzToolsFramework
+{
+    struct DeferredTextboxLikeEdit_default_traits
+    {
+        template<class C>
+        using value_type = decltype(AZStd::declval<const C&>().value());
+
+        template<class C>
+        static auto get(const C* obj)
+        {
+            return obj->value();
+        }
+
+        template<class C, class T>
+        static void set(C* obj, const T& val)
+        {
+            obj->setValue(val);
+        }
+
+        template<class C>
+        static bool isUserEditing(const C* obj)
+        {
+            return obj->hasFocus();
+        }
+
+        template<class C>
+        static void emitEditingFinished(C* obj)
+        {
+            emit obj->editingFinished();
+        }
+    };
+
+    /*
+    A small helper for deferring `setValue' events while a user is editing a textbox-like property.
+    In these cases, any external `setValue' events shouldn't overwrite what the user enters.
+    */
```

Comment:
nit: Can we please update this to be `//!` style (we're trying not to add any more occurrences of `/**/` style for API comments